### PR TITLE
Added relays into Quickshow

### PR DESCRIPTION
**Pull request by xkenia**
Added relays results into Quickshow

### CODE REVIEW

File: quickshow/src/application.cpp [hunk 1]
```diff
@@ -102,6 +102,8 @@ QVariantMap Application::eventInfo()
 		while(q.next())
 			info[q.value(0).toString().mid(6)] = q.value(1);
 		info["profile"] = profile();
+		//set info about relay
+		info["isRelay"] = (info["disciplineId"] == 5 || info["disciplineId"] == 6) ? true : false;
```
Comment (fvacek) [inline quickshow/src/application.cpp:106]: `info["isRelay"] = (info["disciplineId"] == 5 || info["disciplineId"] == 6) ;`

File: quickshow/src/cellrenderer.cpp [hunk 2]
```diff
@@ -3,6 +3,9 @@
 #include <qf/core/log.h>
 #include <qf/core/utils.h>
 
+#include "application.h"
```
Comment (fvacek) [inline quickshow/src/cellrenderer.cpp:6]: Most specific includes must be be first, at the beginning in this case.

File: quickshow/src/cellrenderer.cpp [hunk 3]
```diff
@@ -180,8 +183,8 @@ ResultsCellRenderer::ResultsCellRenderer(const QSize &size, QWidget *widget)
 	m_cellAttributes.resize(ColumnCount);
 	m_cellAttributes[Position] = CellAttribute{4 * m_scaledLetterWidth, Qt::AlignRight};
 	m_cellAttributes[Name] = CellAttribute{0};
-	m_cellAttributes[Registration] = CellAttribute{7 * m_scaledLetterWidth};
-	m_cellAttributes[Time] = CellAttribute{7 * m_scaledLetterWidth, Qt::AlignRight};
+	m_cellAttributes[Registration] = CellAttribute{5 * m_scaledLetterWidth};
+	m_cellAttributes[Time] = CellAttribute{4 * m_scaledLetterWidth, Qt::AlignRight};
```
Comment (fvacek) [inline quickshow/src/cellrenderer.cpp:187]: What if we start to support msecs?

File: quickshow/src/cellrenderer.cpp [hunk 4]
```diff
@@ -180,8 +183,8 @@ ResultsCellRenderer::ResultsCellRenderer(const QSize &size, QWidget *widget)
 	m_cellAttributes.resize(ColumnCount);
 	m_cellAttributes[Position] = CellAttribute{4 * m_scaledLetterWidth, Qt::AlignRight};
 	m_cellAttributes[Name] = CellAttribute{0};
-	m_cellAttributes[Registration] = CellAttribute{7 * m_scaledLetterWidth};
-	m_cellAttributes[Time] = CellAttribute{7 * m_scaledLetterWidth, Qt::AlignRight};
+	m_cellAttributes[Registration] = CellAttribute{5 * m_scaledLetterWidth};
```
Comment (fvacek) [inline quickshow/src/cellrenderer.cpp:186]: Why 7 is too much?

File: quickshow/src/model.cpp [hunk 5]
```diff
@@ -81,20 +81,26 @@ bool Model::addCategoryToStorage()
...
-		QString qs = qb.toString();
+		QString qs = "select cd.relaylegcount as leg_count from classdefs as cd where cd.classid = {{class_id}} and cd.stageid = {{stage_id}}";
+		qfInfo() << "Get leg count for category:" << qs;
 		qs.replace("{{stage_id}}", QString::number(app->cliOptions()->stage()));
 		qs.replace("{{class_id}}", QString::number(cat_id_to_load));
+		qf::core::sql::Query q = app->execSql(qs);
+		q.next();
+		int leg_count_from_class = q.value("leg_count").toInt();
+		for (int leg_num = 1; leg_num <= leg_count_from_class; leg_num++) {
+	QVariantMap category_map;
```
Comment (fvacek) [inline quickshow/src/model.cpp:96]: indentation

File: quickshow/src/model.cpp [hunk 6]
```diff
@@ -81,20 +81,26 @@ bool Model::addCategoryToStorage()
...
-		qb.select2("classes", "name")
-				//.select2("classdefs", "")
-				.select2("courses", "length, climb")
-				.from("classes")
-				.joinRestricted("classes.id", "classdefs.classId", "classdefs.stageId={{stage_id}}")
-				.join("classdefs.courseId", "courses.id")
-				.where("classes.id={{class_id}}");
-		QString qs = qb.toString();
+		QString qs = "select cd.relaylegcount as leg_count from classdefs as cd where cd.classid = {{class_id}} and cd.stageid = {{stage_id}}";
+		qfInfo() << "Get leg count for category:" << qs;
```
Comment (fvacek) [inline quickshow/src/model.cpp:89]: Use qfDebug for debugging

**Review by fvacek — CHANGES_REQUESTED** [review]

File: quickshow/src/cellrenderer.cpp (hunk 3 above)
Comment (xkenia) [reply quickshow/src/cellrenderer.cpp:187]: There is plenty place for 0.1 and 0.01 of seconds, but OK, I'll return it back.

File: quickshow/src/cellrenderer.cpp (hunk 4 above)
Comment (xkenia) [reply quickshow/src/cellrenderer.cpp:186]: On 4 cols and 100% of font size there is too much space, but before someone will find any possibility how to do it in general, we can leave it on 7.

File: quickshow/src/model.cpp (hunk 6 above)
Comment (xkenia) [reply quickshow/src/model.cpp:89]: fixed

File: quickshow/src/model.cpp (hunk 5 above)
Comment (xkenia) [reply quickshow/src/model.cpp:96]: fixed

File: quickshow/src/cellrenderer.cpp (hunk 2 above)
Comment (xkenia) [reply quickshow/src/cellrenderer.cpp:6]: fixed

File: quickshow/src/application.cpp (hunk 1 above)
Comment (xkenia) [reply quickshow/src/application.cpp:106]: fixed

**Comment by fvacek** [comment]
do you plan propagate this change also to `qe3` branch?

**Pull request merged**